.github/workflows/net-lib-release.yml

```diff
@@ -0,0 +1,395 @@
+        description: "The version being released"
+        required: true
+        type: string
+      unique-id:
+        description: "The unique signing id"
+        required: true
+        type: string
+      project:
+        description: "The project being released"
+        required: true
+        type: string
+      csproj:
+        description: "The csproj file name"
+        required: true
+        type: string
+      test_csproj:
+        description: "The csproj test file name"
+        required: true
+        type: string
+      assembly_name:
+        description: "Assembly File Name for project"
+        required: true
+        type: string
+
+jobs:
+  allowList:
+    uses: ./.github/workflows/validate_user.yml
+  sign:
+    needs: [allowList]
+    strategy:
+      matrix:
+        dotnet-version: ["6.0.x"]
+    runs-on: codebuild-AWS-MPL-Release-${{ github.run_id }}-${{ github.run_attempt }}-ubuntu-5.0-large
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      UNIQUE_ID: ${{inputs.unique-id}}
+      CSPROJ: ${{inputs.csproj}}
+      TEST_CSPROJ: ${{inputs.test_csproj}}
+      ASSEMBLY_NAME: ${{inputs.assembly_name}}
+      FILE_NAME: ${{inputs.assembly_name}}
+    steps:
+      - name: Support longpaths on Git checkout
+        run: |
+          git config --global core.longpaths true
+
+      - uses: actions/checkout@v4
+      # The specification submodule is private so we don't have access, but we don't need
+      # it to verify the Dafny code. Instead we manually pull the submodules we DO need.
+      - run: git submodule update --init libraries
+      - run: git submodule update --init smithy-dafny
+
+      - name: Configure AWS Credentials for Tests
+        uses: aws-actions/configure-aws-credentials@v4
+        if: inputs.project == 'ComAmazonawsKms' || inputs.project == 'ComAmazonawsDynamodb' || inputs.project == 'AwsCryptographicMaterialProviders'
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::370957321024:role/GitHub-CI-MPL-Dafny-Role-us-west-2
+          role-session-name: NetReleaseTests
+
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      - name: Setup .NET Core SDK ${{ matrix.dotnet-version }}
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+      - name: Setup Dafny
+        uses: dafny-lang/setup-dafny-action@v1.7.0
+        with:
+          dafny-version: ${{ inputs.dafny }}
+      - name: Export signing ID
+        shell: bash
+        run: |
+          export UNIQUE_ID=${{inputs.unique-id}};
+      - name: make ${{inputs.project}}
+        shell: bash
+        working-directory: ./${{inputs.project}}
+        run: |
+          make transpile_net
+          make test_net FRAMEWORK=net6.0
+      - name: Unset AWS Creds
+        shell: bash
+        run: |
+          echo "AWS_ACCESS_KEY_ID=" >> $GITHUB_ENV
```

Comment:
Question: `sh` has the `unset` command.
Does GitHub env not have an allegory?

Reply:
yes, but in gha land `unset` will only work for that step in the job. It won't carry over to the next step.